osm.py

```diff
@@ -223,6 +220,34 @@ def copy_settings(dest, src, clean_first):
     for item in ITEMS_TO_COPY:
         copy_settings_item(suffix, src, dest, item)
 
+def diff_settings(dest, src):
+    '''
+    Diff the settings between src and dest that would be updated if udpate were used.
+
+    Note that the diffs are done between dest and src to show src as "the new" stuff.
+    '''
+    src = Path(src)
+    dest = Path(dest)
+    # don't compre to self
+    if src.samefile(dest):
+        return
+
+    print(f"Diffing '{dest}' configuration to '{src}'")
+
+    src = src / '.obsidian'
+    dest = dest / '.obsidian'
+
+    for item in ITEMS_TO_COPY:
+        dest_item = dest / item
+        src_item = src / item
+        if not src_item.exists():
+            continue
+        diff_cmd = [DIFF_CMD, "-r", dest_item, src_item]
```

Comment:
Note that on Linux `-r` works even for files.
If this doesn't work on a Mac then we can make it conditional.
Also note that by using the command list form here, the extra shell process isn't needed and more importantly we avoid having to quote paths with spaces, or other funny shell characters.
For the execute command we do want the shell involved because it is a user typed command, not one predefined like this one.

Reply:
On my Mac, the `-r` makes it so it only diffs dirs (and subdirs), not files. So no file diff is ever output afaict.

Maybe related, I have a number of vaults, and they have different sets of directories in them. I get a bunch of things on stderr, e.g.:

```
... /snippets: No such file or directory
... /plugins: No such file or directory
```

etc.

Because of the buffering stuff, all these errors show up in one bunch in the middle of all the stdout.

So, I guess my wish is:

- files get diffed, as well as dirs/subdirs, if that was the intent you were going for (I could play around with different diffs, maybe I just need to install something from homebrew)
- that stderr output gets suppressed (i think?)

Reply:
The buffering was fixed in a later commit, and I see you fixed my typos there, thank you!
As to the `-r` I don't have a mac any more so I don't know the installed diff,
but making it only use `-r` on directories is fine w/me.
Also, I will fix the `No such fie or directory` and report better when one side of the diff doesn't exist,
my own test cases locally didn't catch that, thanks!

Reply:
I think the diff issues have been fixed, but wordsmithing the new messages is welcome!